src/pages/components/map-related/useRecenterOnDataChange.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from 'react'
+import { LatLngTuple } from 'leaflet'
+import { useMap } from 'react-leaflet'
+import { MapProps, position } from './MapWithLocationsAndPath'
+
+export function useRecenterOnDataChange({ positions, plannedRouteStops }: MapProps) {
+  const map = useMap()
+
+  useEffect(() => {
+    const positionsSum = positions.reduce(
+      (acc, { loc }) => [acc[0] + loc[0], acc[1] + loc[1]],
+      [0, 0],
+    )
+    const mean: LatLngTuple = [
+      positionsSum[0] / positions.length || position.loc[0],
+      positionsSum[1] / positions.length || position.loc[1],
+    ]
+
+    map.setView(mean, map.getZoom(), { animate: true })
+  }, [positions, plannedRouteStops])
```

Comment:
this implementation will make `map.setView` to be called every time the `positions` / `plannedRouteStops` are changed.

1. I think we should consider the `plannedRouteStops` when calculating the center
2. I think we should only reset the map in case that the center got changed

so maybe we should calculate `const positionsSum` and the `mean` outside the `useEffect`. 

WDYT?